infra/gcp/istio-prow-private/storage.tf

```diff
@@ -0,0 +1,16 @@
+# Mirrors the gs://istio-build bucket
+resource "google_storage_bucket" "istio_build_private" {
+  name          = "istio-build-private"
+  location      = "US"
+  storage_class = "STANDARD"
+
+  uniform_bucket_level_access = true
```

Comment:
I recommend looking into retention policies, because I dont think buckets can use the TF lifecycle prevent delete

Reply:
I think you cannot delete a bucket with items unless you explicitly allow it:

```
[force_destroy](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/storage_bucket#force_destroy) - (Optional, Default: false) When deleting a bucket, this boolean option will delete all contained objects. If you try to delete a bucket that contains objects, Terraform will fail that run.
```